api/extension-guides/language-model-tutorial.md

```diff
@@ -9,13 +9,324 @@ MetaDescription: Tutorial that walks you through creating a VS Code extension th
+   }
+   catch (e) {
+    // do nothing
+   }
+  }
+ }
+}
+```
+
+Now we can call this method from the `code-tutor.annotate` command. Modify the implementation of the command so that it looks like this:
+
+```ts
+const disposable = vscode.commands.registerTextEditorCommand('code-tutor.annotate', async (textEditor: vscode.TextEditor) => {
+
+  // Get the code with line numbers from the current editor
+  const codeWithLineNumbers = getCodeWithLineNumbers(textEditor);
+
+  // select the 4o chat model
+  let [model] = await vscode.lm.selectChatModels({
+    vendor: 'copilot',
+    family: 'gpt-4o',
+  });
+
+  // init the chat message
+  const messages = [
+    vscode.LanguageModelChatMessage.User(ANNOTATION_PROMPT),
+    vscode.LanguageModelChatMessage.User(codeWithLineNumbers),
+  ];
+
+  // make sure the model is ready
+  if (model) {
+
+    // send the messages array to the model and get the response
+    let chatResponse = await model.sendRequest(messages, {}, new vscode.CancellationTokenSource().token);
+
+    // parse the response and display the annotations
+    await parseChatResponse(chatResponse, textEditor);
+
+  }
+});
+```
+
+We need one last method to actually display the annotations. VS Code calls these "decorations". Add the following method above the `parseChatResponse` method in the `extension.ts` file.
+
+```ts
+function applyDecoration(editor: vscode.TextEditor, line: number, suggestion: string) {
+
+ const decorationType = vscode.window.createTextEditorDecorationType({
+  after: {
+   contentText: ` ${suggestion.substring(0, 25) + "..."}`,
+   color: "grey",
+  },
+ });
+
+ // get the end of the line with the specified line number
+ const lineLength = editor.document.lineAt(line - 1).text.length;
+ const range = new vscode.Range(
+  new vscode.Position(line - 1, lineLength),
+  new vscode.Position(line - 1, lineLength),
+ );
+
+ const decoration = { range: range, hoverMessage: suggestion };
+
+ vscode.window.activeTextEditor?.setDecorations(decorationType, [
+  decoration,
+ ]);
+}
+```
+
+This method takes in our parsed annotation from the model and uses it to create a decoration. This is done by first creating a `TextEditorDecorationType` that specifies the appearance of the decoration. In this case, we are just adding a grey annotation and truncating it to 25 characters. We'll show the full message when the user hovers over the message.
+
+We are then setting where the decoration should appear. We need it to be on the line number that was specified in the annotation, and at the end of the line.
+
+Finally, we set the decoration on the active text editor which is what causes the annotation to appear in the editor.
+
+And that's it! You can now run the extension and open a code file. When you click on the "Tutor" button or select the option from the Command Palette, you should see the code annotations appear in the editor.
 
-## Step 2: do something
+<!-- TODO: Completed project image -->
+![]()
 
```

Comment:
Add a Next steps section to:
- recap what you learned in this tutorial
- provide inspiration for improving this sample or building other extensions